db/blip_handler.go

```diff
@@ -667,7 +673,12 @@ func (bh *blipHandler) handleChanges(rq *blip.Message) error {
 
 			// skip parsing seqno if we're not going to use it (no callback defined)
 			if bh.sgr2PullAddExpectedSeqsCallback != nil {
-				expectedSeqs[IDAndRev{DocID: docID, RevID: revID}] = seqStr(change[0])
+				seq, err := ParseJSONSequenceID(seqStr(change[0]))
+				if err != nil {
+					// We've already asked for the doc/rev for the sequence so assume we're going to receive it... Just log this and carry on
+					base.WarnfCtx(bh.loggingCtx, "Unable to parse expected sequence %q for %q/%q: %v", change[0], base.UD(docID), revID, err)
+				}
+				expectedSeqs[IDAndRev{DocID: docID, RevID: revID}] = seq
```

Comment:
Going back and re-reading this, it means we'll be storing `{Seq:0}` as an expected sequence for a given doc/rev which sounds like a bad idea 🤔

It's possible to move this sequence parsing up above the callbacks, so that we don't request the revision at all in that case, which I think is preferable.

Reply:
@adamcfraser based on discussion I think we can leave the warning as-is, but don't populate expectedSeqs (meaning invalid sequences remain in the processed map when recieved, as we weren't expecting it)